framework/src/userobject/ElementSubdomainModifier.C

```diff
@@ -0,0 +1,353 @@
+    for (const auto & pr : received_nodes)
+      mesh.getMesh().get_boundary_info().remove_node(mesh.getMesh().node_ptr(pr),
+                                                     _moving_boundary_id);
+  };
+
+  Parallel::push_parallel_vector_data(
+      mesh.getMesh().get_boundary_info().comm(), nodes_to_push, node_action_functor);
+}
+
+void
+ElementSubdomainModifier::buildMovedElemsRange()
+{
+  // Clear the object first
+  _moved_elems_range.reset();
+
+  // Make some fake element iterators defining this vector of elements
+  Elem * const * elem_itr_begin = const_cast<Elem * const *>(_moved_elems.data());
+  Elem * const * elem_itr_end = elem_itr_begin + _moved_elems.size();
+
+  const auto elems_begin = MeshBase::const_element_iterator(
+      elem_itr_begin, elem_itr_end, Predicates::NotNull<Elem * const *>());
+  const auto elems_end = MeshBase::const_element_iterator(
+      elem_itr_end, elem_itr_end, Predicates::NotNull<Elem * const *>());
+
+  _moved_elems_range = libmesh_make_unique<ConstElemRange>(elems_begin, elems_end);
+}
+
+void
+ElementSubdomainModifier::buildMovedBndNodesRange()
+{
+  // This is more involved than building the element range, because not all moved nodes are
+  // necessarily associated with a boundary initial condition. We need to first build a set of
+  // boundary nodes. Clear the object first:
+  _moved_bnd_nodes_range.reset();
+
+  // create a vector of the newly activated nodes
+  std::set<const BndNode *> moved_bnd_nodes_set;
+  for (auto & bnd_node : *_mesh.getBoundaryNodeRange())
+  {
+    dof_id_type bnd_node_id = bnd_node->_node->id();
+    if (_moved_nodes.find(bnd_node_id) != _moved_nodes.end())
+      moved_bnd_nodes_set.insert(bnd_node);
+  }
+
+  // Dump all the boundary nodes into a vector so that we can build a range out of it
+  std::vector<const BndNode *> moved_bnd_nodes;
+  moved_bnd_nodes.assign(moved_bnd_nodes_set.begin(), moved_bnd_nodes_set.end());
+
+  // Make some fake node iterators defining this vector of nodes
+  BndNode * const * bnd_node_itr_begin = const_cast<BndNode * const *>(moved_bnd_nodes.data());
+  BndNode * const * bnd_node_itr_end = bnd_node_itr_begin + moved_bnd_nodes.size();
+
+  const auto bnd_nodes_begin = MooseMesh::const_bnd_node_iterator(
+      bnd_node_itr_begin, bnd_node_itr_end, Predicates::NotNull<const BndNode * const *>());
+  const auto bnd_nodes_end = MooseMesh::const_bnd_node_iterator(
+      bnd_node_itr_end, bnd_node_itr_end, Predicates::NotNull<const BndNode * const *>());
+
+  _moved_bnd_nodes_range = libmesh_make_unique<ConstBndNodeRange>(bnd_nodes_begin, bnd_nodes_end);
+}
+
+void
+ElementSubdomainModifier::setSolutionForMovedNodes(SystemBase & sys)
+{
+  ConstBndNodeRange & bnd_node_range = movedBndNodesRange();
+
+  NumericVector<Number> & current_solution = *sys.system().current_local_solution;
+  NumericVector<Number> & old_solution = sys.solutionOld();
+  NumericVector<Number> & older_solution = sys.solutionOlder();
```

Comment:
`ElementSubdomainModifier` has to use with a transient executioner?  If not, we need to check if we need to use old and older solutions

Reply:
How do I check that?

Reply:
You may always need a transient? The element activation won't happen for a steady simulation. 

If this assumption is right, we could merge the code as is

Reply:
This is supposed to work in steady as well. I fixed it.